vms/platformvm/api/static_service.go

```diff
@@ -98,8 +98,16 @@ type Staker struct {
 	StakeAmount *json.Uint64 `json:"stakeAmount,omitempty"`
 }
 
-// GenesisValidator should to be used for genesis validators only.
-type GenesisValidator Staker
+// GenesisValidator is very similar to Staker, but it explicitly uses a ShortNodeID
+// GenesisValidator should be used to generate genesis content only.
+// APIs should use Staker struct only
+type GenesisValidator struct {
+	TxID      ids.ID          `json:"txID"`
+	StartTime json.Uint64     `json:"startTime"`
+	EndTime   json.Uint64     `json:"endTime"`
+	Weight    json.Uint64     `json:"weight"`
+	NodeID    ids.ShortNodeID `json:"nodeID"`
```

Comment:
GenesisValidators will stay with the ShortNodeID. Any other Validator will pick the brand new NodeID